fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatAuthenticationConfig.java

```diff
@@ -143,6 +144,8 @@ protected void configure(HttpSecurity http) throws Exception {
           .exceptionHandling()
           .and()
           .anonymous()
+          // match the same anonymous userid as expected elsewhere
```

Comment:
How is this different from `.anonymous()` above?  Can you write a test that demonstrates this?

Reply:
The default settings in this DSL set the principal to the string `anonymousUser`. The DSL is tricky to test since it's mostly a DSL for setting up beans.

Reply:
So using the updated Spring Security DSL, this whole method would look more like the following:

```java
      http.servletApi(Customizer.withDefaults())
          .exceptionHandling(Customizer.withDefaults())
          .anonymous(anonymous -> anonymous.principal(SpinnakerUsers.ANONYMOUS))
          .addFilterBefore(
              new FiatAuthenticationFilter(fiatStatus, authenticationConverter),
              AnonymousAuthenticationFilter.class);
```

Reply:
I didn't see before. `.principal(SpinnakerUsers.ANONYMOUS)` modifies how `.anonymous()` works...it's not another mechanism in addition to it.

Reply:
I'll figure out some sort of test.

Reply:
Turns out testing this is futile. I've extracted some constants to make this more consistent.